common/djangoapps/student/tests/test_create_account.py

```diff
@@ -561,34 +561,66 @@ def test_email_pattern_requirements(self, email, expect_success):
         else:
             self.assert_error(params, "email", "Unauthorized email address.")
 
-    ## Note andrey.lykhoman:  fix this test
     @ddt.data(
         ('false.result@sorry.bad', True, False),
         ('true.result@its.good', False, True),
     )
     @ddt.unpack
-    def test_email_uniqueness(self, email, is_avaliable, expect_success):
-
-
-        returning = MagicMock()
-        returning.exists = MagicMock(return_value=is_avaliable)
-        returning2 = MagicMock()
-        returning3 = MagicMock()
-        returning3.filter.return_value = returning
-        returning3.exclude.return_value = returning2
-        #returning.get.return_value = returning
-
-
-        with patch('django.contrib.auth.models.User.objects.exists', return_value=is_avaliable):
-            z = UserCreationFormExtended({'email': email})
-            errors = z.errors.as_data()
-            result = True
+    def test_email_uniqueness_create_form(self, email, is_avaliable, expect_success):
+        """
+        Testing uniqueness email in creation user in admin page.
+        You can't create user with certain email address if another user has this email address.
+        If you will try that you must have validation error.
+
+        :param email:
+        :param is_avaliable:
+        :param expect_success:
+        :return:
+        """
+        mockObject = MagicMock()
+        mockObject.exists.return_value = is_avaliable
+        mockObject.filter.return_value = mockObject
+        mockObject.exclude.return_value = mockObject
+        with patch('student.admin.User.objects', mockObject):
+            # Testing email uniqueness in creation user
+            creation_form = UserCreationFormExtended({'email': email})
+            errors = creation_form.errors.as_data()
+            result_creation_form = True
             if 'email' in errors:
-                result = False
-            self.assertEqual(result, expect_success)
-            #UserCreationFormExtended, UserChangeFormExtended
+                result_creation_form = False
+            self.assertEqual(result_creation_form, expect_success)
+
+    @ddt.data(
+        ('false.result@sorry.bad', True, False),
+        ('true.result@its.good', False, True),
+    )
+    @ddt.unpack
+    def test_email_uniqueness_change_form(self, email, is_avaliable, expect_success):
+        """
+        Testing uniqueness email in change user in admin page.
+        You can't change user's email write certain email address if another user has this email address.
+        If you will try that you must have validation error.
+
+        :param email:
+        :param is_avaliable:
+        :param expect_success:
+        :return:
+        """
+        mockObject = MagicMock()
```

Comment:
It looks like code duplication I would move it outside the tests method and insert as return_value for required context manager or decorator `patch`

Reply:
I wrote decorator  - https://github.com/raccoongang/edx-platform/pull/327/files#diff-db937b02d16c28ef5ae9fff01fc1a72bR35